head/src/main/kotlin/io/qalipsis/core/head/report/DataSeriesServiceImpl.kt

```diff
@@ -196,4 +204,48 @@ internal class DataSeriesServiceImpl(
         )
     }
 
+    override suspend fun refresh() {
+        backgroundScope.launch {
+            var pageable = Pageable.from(0, SIZE)
+            var pagedDataSeries = dataSeriesRepository.findDataSeries(pageable)
+            handleQueryUpdate(pagedDataSeries)
+            while (pagedDataSeries.pageNumber < pagedDataSeries.totalPages - 1) {
```

Comment:
I would rather use a `do{} while()` so that you do not have extract work for the first query.